### ISSUE TITLE
Segmentation fault from method `.bridges()`

### ISSUE THREAD

**Issue by adamoppenheimer** [igraph/python-igraph#563]
**Describe the bug**
When `.bridges()` is run on very large datasets, it causes a segmentation fault.

**To reproduce**
I apologize since I don't know how to simply generate a large graph, but I will be happy to go through the details of the following code, and also explain how to extract the graph used in the code.

This code uses the method `.bridges()` and causes a segmentation fault when I run it via ipython3 (I learned about this error from [this issue](https://github.com/tlamadon/bipartitepandas/issues/14#issue-1346536552)):

```python
import numpy as np
import bipartitepandas as bpd

sim_params = bpd.sim_params({'n_workers': 1000000, 'firm_size': 10, 'p_move': 0.05})
clean_params = bpd.clean_params({'connectedness': 'leave_out_match', 'collapse_at_connectedness_measure': True, 'drop_returns': 'returners', 'copy': False, 'verbose': False})
rng = np.random.default_rng(1234)

sim_data = bpd.SimBipartite(sim_params).simulate(rng)

bdf = bpd.BipartiteDataFrame(sim_data).clean(clean_params)
```

Alternatively, I can run this code, which works on large datasets (and gives the exact same results as the first on small datasets where the first doesn't crash), but uses the method `.biconnected_components()` and then takes the subset of edges with size 1 (which is similar to what we discussed in #547):
```python
import numpy as np
import bipartitepandas as bpd

sim_params = bpd.sim_params({'n_workers': 1000000, 'firm_size': 10, 'p_move': 0.05})
clean_params = bpd.clean_params({'connectedness': 'leave_out_match', 'collapse_at_connectedness_measure': False, 'drop_returns': 'returners', 'copy': False, 'verbose': False})
rng = np.random.default_rng(1234)

sim_data = bpd.SimBipartite(sim_params).simulate(rng)

bdf = bpd.BipartiteDataFrame(sim_data).clean(clean_params).collapse()
```

**Version information**
igraph v0.9.11, downloaded via Pip

Thank you for your help!

**Comment by szhorvat** [comment]
The linked issues says that the error message was: "Windows fatal exception: stack overflow"

This is likely the same as this: https://github.com/igraph/igraph/issues/1336

**Comment by szhorvat** [comment]
`igraph_bridges()` has now been refactored to avoid recursive calls. It should no longer crash with stack overflow on large graphs. This improvement will be released with the next version of python-igraph, perhaps in 2-3 weeks.

**Comment by szhorvat** [comment]
@adamoppenheimer You can test the fix right now yourself if you compile python-igraph from source. Instructions are here: https://github.com/igraph/python-igraph/#compiling-the-development-version  The `master` branch contains the 0.9 series and the `develop` branch the soon-to-be-released 0.10 series. Both have this fix.

If you could test and report back that would be much appreciated!

**Comment by adamoppenheimer** [comment]
I apologize that it took me a while to respond!

At first I tried installing from both the master branch and dev branch and they still both gave a segfault on my code.

However, then I tried uninstalling igraph before compiling from source. This was more work than I expected, since I had to install cmake! But once it installed, I tried out both branches and they work!

Thank you for the fast work, I really appreciate it! I also want to thank you for developing this package in general, it is excellent! I was originally using networkx and it took over 24 hours on some code (I ended up stopping it since it was taking so long) that only takes around 7 minutes with your package (and it's even faster now since I've changed to different igraph methods like `.bridges()`)!
